jax/interpreters/pxla.py

```diff
@@ -621,13 +624,15 @@ def xla_pmap_impl(fun: lu.WrappedFun, *args, backend, axis_name, axis_size,
                                    in_axes, out_axes_thunk,
                                    donated_invars, global_arg_shapes,
                                    *abstract_args)
+
   # Don't re-abstractify args unless logging is enabled for performance.
   if config.jax_distributed_debug:
     distributed_debug_log(("Running pmapped function", name),
                           ("python function", fun.f),
                           ("devices", devices),
-                          ("abstract args", map(xla.abstractify, args)))
-  return compiled_fun(*args)
+                          ("abstract args", map(xla.abstractify, args)),
+                          ("Fingerprint", compiled_fun[1].fingerprint))
+  return compiled_fun[0](*args)
```

Comment:
Instead of using `compiled_fun[0]` and `compiled_fun[1]`, unpack these into two separate variables immediately at the `parallel_callable` call site with more descriptive names. Something like:
```
compiled_fun, xla_executable = parallel_callable(...)
```
(These still aren't the most descriptive names, but it's the best I can come up with)